components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java

```diff
@@ -122,6 +155,52 @@ public SAMLSSOServiceProviderDTO addSAMLServiceProvider(SAMLSSOServiceProviderDT
         }
     }
 
+    private static Map<String, Object> buildSPData(SAMLSSOServiceProviderDO app) {
+
+        if (app == null) {
+            return new HashMap<>();
+        }
+
+        Gson gson = new Gson();
+        String json = gson.toJson(app);
+        return gson.fromJson(json, new TypeToken<Map<String, Object>>() {
+        }.getType());
+    }
+
+    /**
+     * This method is used to retrieve logged in tenant domain.
+     * @return logged in tenant domain.
+     */
+    private String getLoggedInTenantDomain() {
+
+        if (!IdentityTenantUtil.isTenantedSessionsEnabled()) {
+            return SAMLSSOUtil.getTenantDomainFromThreadLocal();
+        }
+
+        return IdentityTenantUtil.getTenantDomainFromContext();
+
+    }
+
+    private AuthenticatedUser getLoggedInUser(String tenantDomain) {
+
+        String tenantAwareLoggedInUsername = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        AuthenticatedUser loggedInUser = null;
+        if (StringUtils.isNotEmpty(tenantAwareLoggedInUsername)) {
+            loggedInUser = buildAuthenticatedUser(tenantAwareLoggedInUsername, tenantDomain);
+        }
+        return loggedInUser;
+
+    }
+
+    AuthenticatedUser buildAuthenticatedUser(String tenantAwareUser, String tenantDomain) {
```

Comment:
any reason not to define an access modifier?

Reply:
Fixed